### PR TITLE
fix: sidebar expand/collapse status error

### PR DESCRIPTION
**Pull request by FeiWang1119**
diff expand/collapse from sidebar visible

Issue: https://github.com/linuxdeepin/developer-center/issues/5297

### CODE REVIEW

**Review by kegechen — COMMENTED** [review]
+1

File: src/widgets/dtitlebar.cpp [hunk 1]
```diff
@@ -1293,6 +1293,7 @@ void DTitlebar::setSidebarHelper(DSidebarHelper *helper)
 
     connect(helper, &DSidebarHelper::visibleChanged, this, [this](bool visible){
         d_func()->expandButton->setVisible(visible);
+        d_func()->sidebarBackgroundWidget->setVisible(visible ? d_func()->sidebarHelper->expanded() : false);
```
Comment (18202781743) [inline src/widgets/dtitlebar.cpp:1296]: 这个fallback的函数可以放到sidebarhelper类中，

**Review by 18202781743 — APPROVED** [review]

File: src/widgets/private/dmainwindow_p.h [hunk 2]
```diff
@@ -68,6 +68,14 @@ class DSidebarHelper : public QObject
         Q_EMIT expandChanged(m_expanded);
     }
 
+    inline bool sectionVisible() const
+    {
+        if (m_visible)
+            return m_expanded;
+
+        return m_visible;
```
Comment (kegechen) [inline src/widgets/private/dmainwindow_p.h:76]: return m_visible ? m_expanded : false;

**Review by kegechen — APPROVED** [review]

File: src/widgets/private/dmainwindow_p.h [hunk 3]
```diff
@@ -68,6 +68,11 @@ class DSidebarHelper : public QObject
         Q_EMIT expandChanged(m_expanded);
     }
 
+    inline bool sectionVisible() const
```
Comment (kegechen) [inline src/widgets/private/dmainwindow_p.h:71]: 要不叫sidebarwidgetvisible

File: src/widgets/private/dmainwindow_p.h (hunk 2 above)
Comment (FeiWang1119) [reply src/widgets/private/dmainwindow_p.h]: done

**Pull request merged**